### ISSUE TITLE
repaint is not working with multiline prompts

### ISSUE THREAD

**Issue by BarbzYHOOL** [fish-shell/fish-shell#5033]
fish 2.7

How to reproduce?

- install a multiline prompt/theme like cmorrell or kawasaki (so it means you have one line with a lot of information, and then a second line to type your commands)
- type "cat" but do not press enter
- press "Alt+w" and depending on the free space of your first line, you will see some words or none at all

With a default theme (one line), you would see under a your command line:

    cat (1)              - concatenate files and print on the standard output

How to solve this? Really annoying

**Comment by mqudsi** [comment]
The default binding for `alt-w`:

```fish
bind \ew 'set tok (commandline -pt); if test $tok[1]; echo; whatis $tok[1]; commandline -f repaint; end'
```

You could probably just add another `echo` in there to make it work:

```fish
bind \ew 'set tok (commandline -pt); if test $tok[1]; echo; echo; whatis $tok[1]; commandline -f repaint; end'
```

**Comment by BarbzYHOOL** [comment]
Funny I spent several hours trying to fix this today.

No, it doesn't work. But I added this instead:

        bind \ew 'set tok (commandline -pt); if test $tok[1]; printf "\n"; whatis $tok[1]; and commandline -f kill-whole-line; and commandline -f execute; end'

It's hacky at best.

But it displays twice fish_right_prompt (which shows twice the date for me) it looks like a bug but I don't know how to avoid it. The only way would be to not display the date if previous command was empty or something like that, but no idea how.
